todo_app/app.py

```diff
@@ -41,37 +53,55 @@ def validate_user():
                 user_details = requests.get('https://api.github.com/user', headers=headers_for_user).json()
                 id = User(user_details['id'])
                 login_user(id)
+                app.logger.info("User ID %s successfully logged in", id)
                 return redirect("/")
 
         def authorized():
-                return ((current_user.role == 'writer') or (app.config['LOGIN_DISABLED'] == True))                         
+                return ((app.config['LOGIN_DISABLED'] == True) or (current_user.role == 'writer'))                         
         
         @app.route('/')
         @login_required
         def index():
-                task_view_model = ViewModel(get_items())
-                auth_type = None if app.config['LOGIN_DISABLED'] == True else current_user.role
+                try:
+                        task_view_model = ViewModel(get_items())
+                except:
+                        app.logger.critical("To do list data failed to generate")            
```

Comment:
This would arguably make more sense as an "error" rather than a "critical"; critical typically means our application has entirely fallen over. However I can see you've used that elsewhere below, and an argument could be made that the homepage not loading makes the app unusable so functionally the same - though I'd be weary of one failed db connection which is immediately resolved on refresh triggering a critical log!